ledger/participant-integration-api/src/test/suite/scala/platform/store/ConversionsSpec.scala

```diff
@@ -3,54 +3,165 @@
 
 package com.daml.platform.store
 
+import com.daml.error.{
+  ContextualizedErrorLogger,
+  DamlContextualizedErrorLogger,
+  ErrorCodesVersionSwitcher,
+}
 import com.daml.ledger.api.domain
+import com.daml.ledger.api.domain.RejectionReason
+import com.daml.lf.data.Ref
+import com.daml.lf.transaction.GlobalKey
+import com.daml.lf.value.Value
+import com.daml.lf.value.Value.ValueText
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.platform.server.api.validation.ErrorFactories
 import com.daml.platform.store.Conversions._
 import io.grpc.Status
+import org.scalatest.Assertion
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AsyncWordSpec
 
+import scala.annotation.nowarn
+
+@nowarn("msg=deprecated")
 class ConversionsSpec extends AsyncWordSpec with Matchers {
+  private implicit val contextualizedErrorLogger: ContextualizedErrorLogger =
+    new DamlContextualizedErrorLogger(
+      ContextualizedLogger.get(getClass),
+      LoggingContext.ForTesting,
+      None,
+    )
+
   "converting rejection reasons" should {
     "convert an 'Inconsistent' rejection reason" in {
-      val reason = domain.RejectionReason.Inconsistent("This was not very consistent.")
-      val converted = reason.toParticipantStateRejectionReason
-      converted.code should be(Status.Code.ABORTED.value())
-      converted.message should be("Inconsistent: This was not very consistent.")
+      assertConversion(domain.RejectionReason.Inconsistent("This was not very consistent."))(
+        v1_expectedCode = Status.Code.ABORTED.value(),
+        v1_expectedMessage = "Inconsistent: This was not very consistent.",
+        v2_expectedCode = Status.Code.FAILED_PRECONDITION.value(),
+        // TODO error codes: remove context from end of message (see [[com.daml.error.definitions.TransactionError.rpcStatus]])
+        v2_expectedMessage = "INCONSISTENT(9,0): Inconsistent: This was not very consistent.",
+      )
+    }
+
+    "convert an 'ContractsNotFound' rejection reason" in {
+      assertConversion(domain.RejectionReason.ContractsNotFound(Set("cId_1", "cId_2")))(
+        v1_expectedCode = Status.Code.ABORTED.value(),
+        v1_expectedMessage = "Inconsistent: Could not lookup contracts: [cId_1, cId_2]",
+        v2_expectedCode = Status.Code.NOT_FOUND.value(),
+        // TODO error codes: remove context from end of message (see [[com.daml.error.definitions.TransactionError.rpcStatus]])
```

Comment:
To improve the chances of these getting fixed, I'd suggest tracking such TODOs via an issue tracker instead.

Reply:
Actually this one I have already fixed in this PR but forgot to delete the TODO. I'll remove